project/CoreJVMTest.scala

```diff
@@ -0,0 +1,242 @@
+import scala.util.Random
+
+import sbt._
+import Keys._
+
+object CoreJVMTest {
+  lazy val testsGenerator = Def.task[Seq[File]] {
+    val managed = (Test / sourceManaged).value
+
+    generateEnumTests(managed) ++ generateValueEnumTest[Int](
+      managed,
+      Iterator.continually(Random.nextInt())
+    ) ++ generateValueEnumTest[Long](
+      managed,
+      Iterator.continually(Random.nextLong()),
+      valueSuffix = "L"
+    ) ++ generateValueEnumTest[Short](
+      managed,
+      Iterator
+        .continually(Random.nextInt(Short.MaxValue - Short.MinValue) + Short.MinValue)
+        .map(_.toShort)
+    ) ++ generateValueEnumTest[Byte](
+      managed,
+      Iterator
+        .continually(Random.nextInt(Byte.MaxValue - Byte.MinValue) + Byte.MinValue)
+        .map(_.toByte)
+    ) ++ generateValueEnumTest[String](
+      managed,
+      Random.alphanumeric.grouped(10).map(_.mkString),
+      "\"",
+      "\""
+    ) ++ generateValueEnumTest[Char](
+      managed,
+      Iterator.continually(Random.nextPrintableChar()).filter(c => Character.isAlphabetic(c.toInt)),
+      "'",
+      "'"
+    )
+
+  }
+
+  private def generateEnumTests(outdir: File): Seq[File] = {
+    val bf = outdir / "EnumBaseSpec.scala"
+
+    IO.writer[Seq[File]](bf, "", IO.defaultCharset, false) { w0 =>
+      w0.append("""package generated
+
+trait EnumBaseSpec
+  extends org.scalatest.funspec.AnyFunSpec 
+  with org.scalatest.matchers.should.Matchers
+""")
+
+      val res = (1 to 100).flatMap { i =>
+        val enumName = s"Enum${i}"
+
+        val ef = outdir / s"${enumName}.scala"
+
+        IO.writer[Seq[File]](ef, "", IO.defaultCharset, false) { w1 =>
+          // Generate enum file
+
+          w1.append(s"""package generated
+
+import enumeratum._
+
+sealed trait ${enumName} extends EnumEntry
+
+object ${enumName} extends Enum[${enumName}] {
+  val values = findValues
+
+""")
+
+          val members = Random.shuffle(1 to Random.nextInt(20)).map { n =>
+            val nme = s"Member$n"
+
+            w1.append(s"  case object ${nme} extends ${enumName}\n")
+
+            nme
+          }
+
+          w1.append(s"""}
+""")
+
+          // Generate tests in separate file/trait
+          val tf = outdir / s"${enumName}Test.scala"
```

Comment:
I think it might be sensible to have the file name match the name of the class? I think the value test files are also suffixed with `Spec`

```suggestion
          val tf = outdir / s"${enumName}Spec.scala"
```

Reply:
The associated trait is named `${enumName}Test` (line 94)